pkg/network/protocols/amqp/server.go

```diff
@@ -19,18 +22,67 @@ const (
 	User = "guest"
 	// Pass is the password to use for authentication
 	Pass = "guest"
+
+	// AmqpPlaintext is a flag to indicate that the server should be started with plaintext
+	AmqpPlaintext = false
+	// AmqpTLS is a flag to indicate that the server should be started with TLS
+	AmqpTLS = true
 )
 
-// RunServer runs an amqp server in a docker container
-func RunServer(t testing.TB, serverAddr, serverPort string) error {
-	env := []string{
+// RunServer runs an AMQP server in a docker container.
+func RunServer(t testing.TB, serverAddr, serverPort string, withTLS bool) error {
+	t.Helper()
+
+	var env []string
+	var startupRegexp *regexp.Regexp
+	if withTLS {
+		env, startupRegexp = tlsConfig(t, serverAddr, serverPort)
+	} else {
+		env, startupRegexp = plaintextConfig(t, serverAddr, serverPort)
+	}
+	dir, _ := httpUtils.CurDir()
+
+	return protocolsUtils.RunDockerServer(t, "amqp", dir+"/testdata/docker-compose.yml", env, startupRegexp, protocolsUtils.DefaultTimeout, 3)
+}
+
+// commonConfig returns the common environment variables for the amqp server,
+// independently of whether or not the server uses TLS or not..
+func commonConfig(t testing.TB, serverAddr, serverPort string) []string {
+	t.Helper()
+
+	cert, _, err := httpUtils.GetCertsPaths()
+	require.NoError(t, err)
+	certsDir := filepath.Dir(cert)
+
+	return []string{
 		"AMQP_ADDR=" + serverAddr,
 		"AMQP_PORT=" + serverPort,
 		"USER=" + User,
 		"PASS=" + Pass,
+		"CERTS_PATH=" + certsDir,
```

Comment:
why is it in the common config?

Reply:
That's because of Docker compose lack of conditionals. Originally it was in the TLS config, but that meant `CERTS_PATH` was not set with the plaintext config, leading to a volume mount with an empty source, which prevented docker compose from starting the container.
I felt this was pretty harmless to do. Another solution would be to use a default value for the variable in docker-compose.yml, but what would that default be?